Subject: Tolerate empty EDITOR and VISUAL envvars
Last-Updated: 2010-04-11

Empty EDITOR and VISUAL environment variables (as opposed to unset ones) should
not cause fatal errors.

Bug-Debian: http://bugs.debian.org/148809

Acked-by: Christian Kastner <debian@kvr.at>
Index: patched/crontab.c
===================================================================
--- patched.orig/crontab.c	2010-05-06 18:18:30.213560954 +0200
+++ patched/crontab.c	2010-05-06 18:18:33.711435964 +0200
@@ -388,8 +388,12 @@
 	}
 	mtime = statbuf.st_mtime;
 
-	if ((!(editor = getenv("VISUAL")))
-	 && (!(editor = getenv("EDITOR")))
+
+
+        /* Okay, edit the file */
+
+	if ((!((editor = getenv("VISUAL")) && strlen(editor)))
+	 && (!((editor = getenv("EDITOR")) && strlen(editor)))
 	    ) {
 		editor = EDITOR;
 	}
